src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyWorkerService.java

```diff
@@ -0,0 +1,936 @@
+        notificationService.publishStop("publishNonEvacuatedEnergyStopped-out-0", resultUuid, receiver);
+        LOGGER.info(CANCEL_MESSAGE + " (resultUuid='{}')", resultUuid);
+    }
+
+    @Bean
+    public Consumer<Message<String>> consumeNonEvacuatedEnergyRun() {
+        return message -> {
+            NonEvacuatedEnergyResultContext nonEvacuatedEnergyResultContext = NonEvacuatedEnergyResultContext.fromMessage(message, objectMapper);
+            try {
+                runRequests.add(nonEvacuatedEnergyResultContext.getResultUuid());
+                AtomicReference<Long> startTime = new AtomicReference<>();
+
+                startTime.set(System.nanoTime());
+                String result = run(nonEvacuatedEnergyResultContext.getRunContext(), nonEvacuatedEnergyResultContext.getResultUuid());
+                long nanoTime = System.nanoTime();
+                LOGGER.info("Just run in {}s", TimeUnit.NANOSECONDS.toSeconds(nanoTime - startTime.getAndSet(nanoTime)));
+
+                nonEvacuatedEnergyRepository.insert(nonEvacuatedEnergyResultContext.getResultUuid(), result, SensitivityAnalysisStatus.COMPLETED.name());
+                long finalNanoTime = System.nanoTime();
+                LOGGER.info("Stored in {}s", TimeUnit.NANOSECONDS.toSeconds(finalNanoTime - startTime.getAndSet(finalNanoTime)));
+
+                if (result != null) {  // result available
+                    notificationService.sendResultMessage("publishNonEvacuatedEnergyResult-out-0", nonEvacuatedEnergyResultContext.getResultUuid(), nonEvacuatedEnergyResultContext.getRunContext().getReceiver());
+                    LOGGER.info("Non evacuated energy complete (resultUuid='{}')", nonEvacuatedEnergyResultContext.getResultUuid());
+                } else {  // result not available : stop computation request
+                    if (cancelComputationRequests.get(nonEvacuatedEnergyResultContext.getResultUuid()) != null) {
+                        cleanResultsAndPublishCancel(nonEvacuatedEnergyResultContext.getResultUuid(), cancelComputationRequests.get(nonEvacuatedEnergyResultContext.getResultUuid()).getReceiver());
+                    }
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } catch (Exception | OutOfMemoryError e) {
+                if (!(e instanceof CancellationException)) {
+                    LOGGER.error(FAIL_MESSAGE, e);
+                    notificationService.publishFail("publishNonEvacuatedEnergyFailed-out-0", nonEvacuatedEnergyResultContext.getResultUuid(), nonEvacuatedEnergyResultContext.getRunContext().getReceiver(), e.getMessage(), nonEvacuatedEnergyResultContext.getRunContext().getUserId());
+                    nonEvacuatedEnergyRepository.delete(nonEvacuatedEnergyResultContext.getResultUuid());
+                    nonEvacuatedEnergyRepository.insertStatus(List.of(nonEvacuatedEnergyResultContext.getResultUuid()), SensitivityAnalysisStatus.FAILED.name());
```

Comment:
why do we need this insertStatus? it does not seem to be in the corresponding code of other computation services? (SC, LF...)

Reply:
Yes, but same thing has been done in sensitivity analysis ...